### ISSUE TITLE
Universal Storage Service

### ISSUE THREAD

**Issue by jeffwhelpley** [angular/angular#7303]
We need a universal service (i.e. both on client and server) that can be used by any Angular 2 app to access localStorage or other similar repositories. This issue is for coming up with a design for this service and implementing it in core. Once that is in place, the server side implementation of StorageService (which can use Redis or something else) will be implemented on the Angular Universal repo:

https://github.com/angular/universal/issues/272

**Comment by Binvention** [comment]
it would probably be more appropriate under platform rather than core. because it would not be the same code for browser as it would be for server.

**Comment by manekinekko** [comment]
We will need an implementation for both local and session storage.

**Comment by jeffwhelpley** [comment]
@Binvention yes, you may be right about platform. As far as storage on the server, the implementation would typically either be in memory storage or something like Redis.

**Comment by jeffwhelpley** [comment]
Closing in favor of #7438.
